### ISSUE TITLE
helios-1.7.2 failed to build on x86_64

### ISSUE THREAD

**Issue by nysnatuss** [haikuports/haikuports#1979]
[haiku-apps/helios-1.7.2](https://github.com/haikuports/haikuports/blob/master/haiku-apps/helios/helios-1.7.2.recipe) failed to build on x86_64.

The main error is: ([Full log](https://pastebin.com/ABxZYz8M))
```
error: call of overloaded 'RemoveItem(long int)' is ambiguous

...

note: candidate: bool BMenu::RemoveItem(BMenuItem*)
    bool    RemoveItem(BMenuItem* item);
            ^
note: candidate: BMenuItem* BMenu::RemoveItem(int32)
    BMenuItem*   RemoveItem(int32 index);
                 ^
note: candidate: bool BMenu::RemoveItem(BMenu*)
    bool    RemoveItem(BMenu* menu);
            ^
```
It seems like `RemoveItem()` does not accept long int and throw an error instead. ([Ref](https://stackoverflow.com/questions/4672152/call-of-overloaded-function-is-ambiguous))

**Comment by Oyami-Srk** [comment]
Which version of Haiku are you using? I tested in Haiku 51602 and the atomic_add requires int32(I think in x64, atomic add should requires int64.

**Comment by Oyami-Srk** [comment]
Fixed

**Comment by scottmc** [comment]
Please upstream the patch to HaikuArchives, once that is accepted, then this issue can be closed.

**Comment by nysnatuss** [comment]
Fixed in #1993.
